src/hw/hw_mngr.cpp

```diff
@@ -128,6 +257,58 @@ bool HWManager::setOption(const OptionType& optType, const int& instance,
     return true;
 }
 
+FanState HWManager::getDetectFanState()
+{
+    return detectFansState;
+}
+
+void HWManager::publishSystemFans()
+{
+    if (!config.desc)
+        return;
+
+    //  remove existing system fans before re-publish
+    for (auto it = fans.begin(); it != fans.end(); )
+    {
+        const auto& fan = *it;
+
+        std::string fanName = fan->name();
+        size_t pos = fanName.find("Sys_Fan");
```

Comment:
I would choose to sacrifice the readability to avoid an excess memory allocation.
```cpp
auto pos = fan->name().find("Sys_Fan");
```

Reply:
fixed